Framework/Algorithms/src/PDCalibration.cpp

```diff
@@ -1115,6 +1137,44 @@ API::MatrixWorkspace_sptr PDCalibration::rebin(API::MatrixWorkspace_sptr wksp) {
   return wksp;
 }
 
+std::set<detid_t> PDCalibration::detIdsForTable() {
+  std::set<detid_t> detids;
+
+  // return early since everything is being used
+  if (isDefault("StartWorkspaceIndex") && isDefault("StopWorkspaceIndex"))
+    return detids;
+
+  // get the indices to loop over
+  std::size_t startIndex = 0;
+  if (!isDefault("StartWorkspaceIndex"))
+    startIndex = static_cast<std::size_t>(m_startWorkspaceIndex);
+  std::size_t stopIndex = m_uncalibratedWS->getNumberHistograms();
+  if (!isDefault("StopWorkspaceIndex"))
+    stopIndex = static_cast<std::size_t>(m_stopWorkspaceIndex);
```

Comment:
can `m_stopWorkspaceIndex` and `m_starWorkspaceIndex` from earlier be used here?

Reply:
(and if not, the ternary logic was cooler)

Reply:
Personally, I try not to set member values in the `validateInputs()` method because it is not necessarily where people will look for that behavior.

Reply:
Can you not use `startIndex = m_startWorkspaceIndex` and `stopIndex = m_stopWorkspaceIndex`?

Reply:
I don't think this method is directly related to `validateInputs()`, so you should be able to use `m_startWorkspaceIndex` and `m_stopWorkspaceIndex` directly in the flow look below, unless there's some other logic preventing it

Reply:
`startIndex` is `std::size_t` to make indexing into the Workspace2D easy

`m_startWorkspaceIndex` is `int` to make interacting with `FitPeaks` easy

Unfortunately you need to either cast all the time (don't do that in a loop) or make a local copy with the right type